waiter/src/waiter/scheduler.clj

```diff
@@ -754,7 +754,9 @@
 (defn add-instance-to-transient-store!
   [transient-store service-id instance event-type & args]
   (log-service-instance instance event-type)
-  (add-instance-to-buffered-collection! transient-store service-id instance args))
+  (if (= (count args) 3)
+    (add-instance-to-buffered-collection! transient-store service-id instance (nth args 0) (nth args 1) (nth args 2))
+    (add-instance-to-buffered-collection! transient-store service-id instance (nth args 0))))
```

Comment:
Should I use (first ) or (nth ... 0) in this case? I used nth to preserve the fact that nth is used on the line above.

Reply:
```suggestion
    (apply add-instance-to-buffered-collection! transient-store service-id instance args)))
```